main.cpp

```diff
@@ -563,27 +579,45 @@ int main()
     DoubleType dt ( 2 );
     IntType it ( 2 ) ;
 
-    std::cout << "FloatType add result=" << ft.add( 2.0f ) << std::endl;
-    std::cout << "FloatType subtract result=" << ft.subtract( 2.0f ) << std::endl;
-    std::cout << "FloatType multiply result=" << ft.multiply( 2.0f ) << std::endl;
-    std::cout << "FloatType divide result=" << ft.divide( 16.0f) << std::endl<< std::endl;  
-    
-    
-    std::cout << "DoubleType add result=" << dt.add(2.0) << std::endl;
-    std::cout << "DoubleType subtract result=" << dt.subtract(2.0) << std::endl;
-    std::cout << "DoubleType multiply result=" << dt.multiply(2.0) << std::endl;
-    std::cout << "DoubleType divide result=" << dt.divide(static_cast<double>(5.f) ) << std::endl << std::endl;
+    ft+= ( 2.0f );
+    std::cout << "FloatType add result=" << ft << std::endl;
+    ft-= ( 2.0f );
+    std::cout << "FloatType subtract result=" << ft << std::endl;
+    ft*= ( 2.0f );
+    std::cout << "FloatType multiply result=" << ft << std::endl;
+    ft/= ( 16.0f );
+    std::cout << "FloatType divide result=" << ft << std::endl<< std::endl;  
```

Comment:
Add some space around the operators here.
`it+=` `dt*=` etc are difficult to read.
Help yourself and make the code easier to read
- [ ] revise